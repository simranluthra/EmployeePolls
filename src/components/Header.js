import { Link } from "react-router-dom";
import { useDispatch } from "react-redux";

const navItems = [
  {
    title: "Home",
    link: "/",
  },
  {
    title: "Leaderboard",
    link: "/leaderboard",
  },
  {
    title: "New",
    link: "/add",
  },
];

const Header = (props) => {
  const { currentUser } = props;
  const dispatch = useDispatch();
  const handleLogout = () =>
    dispatch({
      type: "SIGNOUT",
    });

  return (
    <nav
      className="navbar navbar-expand-lg fixed-top navbar-light bg-light"
      aria-label="Main navigation"
    >
      <div className="container-fluid">
        <div className="d-flex justify-content-between">
          <ul className="navbar-nav me-auto mb-2 mb-lg-0">
            {navItems.map((item, index) => {
              return (
                <li className="nav-item" key={`${item.title}-${index}`}>
                  <Link className="nav-link active" to={item.link}>
                    {item.title}
                  </Link>
                </li>
              );
            })}
          </ul>
        </div>
        {currentUser && (
          <span className="">
            <ul className="navbar-nav me-end mb-2 mb-lg-0">
              <li className="nav-item">
                <img
                  src={currentUser.avatarURL}
                  className="img-thumbnail img-fluid me-end"
                  alt={currentUser.name + " avatar"}
                  width="45"
                />
              </li>
              <li className="nav-item">
                <span className="nav-link active">{currentUser.name}</span>
              </li>
              <li className="nav-item">
                <Link to="/" className="nav-link" onClick={handleLogout}>
                  Logout
                </Link>
              </li>
            </ul>
          </span>
        )}
      </div>
    </nav>
  );
};

export default Header;
