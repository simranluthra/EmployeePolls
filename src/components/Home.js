import { useState, useEffect } from "react";
import QuestionRow from "./QuestionRow";
import CardGroup from "react-bootstrap/CardGroup";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Tab from "react-bootstrap/Tab";
import Tabs from "react-bootstrap/Tabs";

const Home = (props) => {
  console.log(props);
  const { authedUser, users, questions } = props;
  const [answeredQuestions, setAnsweredQuestions] = useState([]);
  const [unAnsweredQuestions, setUnAnsweredQuestions] = useState([]);

  useEffect(() => {
    let answered = [];
    let unAnswered = [];
    for (const id in questions) {
      const question = questions[id];
      const voted = question.optionOne.votes
        .concat(question.optionTwo.votes)
        .some((id) => id === authedUser);
      voted ? answered.push(question) : unAnswered.push(question);
    }
    const sorted = (arr) => arr.sort((a, b) => b.timestamp - a.timestamp);
    setAnsweredQuestions(sorted(answered));
    setUnAnsweredQuestions(sorted(unAnswered));
  }, [authedUser, questions]);

  return (
    <main className="container">
      <div className="bg-body rounded shadow-sm">
      <Tabs defaultActiveKey="UnansweredQuestions" id="uncontrolled-tab-example" className="mb-2 mt-5">
        <Tab eventKey="UnansweredQuestions" title="UnansweredQuestions">
        <div className="bg-body rounded shadow-sm">
          <CardGroup>
            <Row className="g-4 w-100">
              <Col>
                  {unAnsweredQuestions.map((question) => {
                    return <QuestionRow key={question.id} question={question} users={users} />;
                  })}
              </Col>
            </Row>
          </CardGroup>
        </div>
        </Tab>
        <Tab eventKey="answeredQuestions" title="answeredQuestions">
              <CardGroup>
                <Row className="g-4 w-100">
                  <Col>
                    {answeredQuestions.map((question) => {
                      return <QuestionRow key={question.id} question={question} users={users} />;
                    })}
                  </Col>
                </Row>
              </CardGroup>
        </Tab>
      </Tabs>
      </div>
    </main>
  );
};

export default Home;
