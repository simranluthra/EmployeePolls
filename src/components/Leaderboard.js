const Leaderboard = (props) => {
  const { users = {} } = props;
  console.log("hello", users);
  const sortedUsers = Object.values(users).sort((userX, userY) => (
      (Object.entries(userY.answers).length + userY.questions.length) -
      (Object.entries(userX.answers).length + userX.questions.length)
    )
  );

  return (
    <div className="container mt-5">
      <table className="table mt-4">
        <thead className="table-light">
          <tr>
            <th scope="col">User</th>
            <th className="text-center" scope="col">
              Answered
            </th>
            <th className="text-center" scope="col">
              Created
            </th>
          </tr>
        </thead>
        <tbody>
          {sortedUsers.map((user) => (
            <tr key={user.id}>
              <td>
                <img
                  src={user.avatarURL}
                  className="img-thumbnail"
                  alt={"username"}
                  width="50"
                />
                {user.name}
              </td>
              <td className="text-center">
                {Object.entries(user.answers).length}
              </td>
              <td className="text-center">{user.questions.length}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Leaderboard;
