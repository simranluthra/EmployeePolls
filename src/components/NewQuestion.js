import { useState, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useNavigate } from "react-router-dom";

import { saveQuestion } from "../redux/actions";

const NewQuestion = () => {
  const { authedUser: author } = useSelector((store) => store);
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const [optionOneText, setOptionOneText] = useState("");
  const [optionTwoText, setOptionTwoText] = useState("");
  const [toHome, setToHome] = useState(false);
  const [statusMessage, setStatusMessage] = useState({
    message: "",
    alertClass: ""
  });

  useEffect(() => {
    setStatusMessage(null);
  }, [optionOneText, optionTwoText]);

  useEffect(() => {
    if (toHome) navigate("/");
  }, [navigate, toHome]);

  function handleSubmit() {
    if (!optionOneText.trim() || !optionTwoText.trim()) {
      return setStatusMessage({
        message: "Options cannot be empty.",
        alertClass: "danger"
      });
    } else {
      const question = {
        optionOneText,
        optionTwoText,
        author
      };
      submit(question);
    }
  }

  function submit(question) {
    setStatusMessage({
      message: "Adding question...",
      alertClass: "info"
    });
    dispatch(saveQuestion(question))
      .then(({ type }) => {
        if (type === "SET_QUESTION") {
          setToHome(true);
        }
      });
  }

  return (
    <div className="container mt-4 text-center">
      <h1>Would You Rather</h1>
      <p>Create your own poll</p>
      <label htmlFor="question1" className="form-label">
        First Option:
      </label>
      <input
        onChange={({ target }) => setOptionOneText(target.value)}
        value={optionOneText}
        type="text"
        className="form-control"
        placeholder="First option.."
        data-testid="option-one"
      />
      <label htmlFor="question2" className="form-label">
        Second Option:
      </label>
      <input
        onChange={({ target }) => setOptionTwoText(target.value)}
        value={optionTwoText}
        type="text"
        className="form-control"
        placeholder="Second option.."
        data-testid="option-two"
      />
      {statusMessage && (
        <div
          data-testid="notify-text"
          className={`my-2 alert alert-${statusMessage.alertClass}`}
          role="alert"
        >
          {statusMessage.message}
        </div>
      )}
      <button
        data-testid="submit-btn"
        type="button"
        className="btn btn-primary my-3"
        onClick={handleSubmit}
      >
        Submit
      </button>
    </div>
  );
};

export default NewQuestion;
