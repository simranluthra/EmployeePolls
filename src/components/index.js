import Header from "./Header";
import Home from "./Home";
import NewQuestion from "./NewQuestion";
import Leaderboard from "./Leaderboard";
import Login from "./Login";
import PageNotFound from "./PageNotFound";
import QuestionDetails from "./QuestionDetails";

export {
  Header,
  Home,
  NewQuestion,
  Leaderboard,
  Login,
  PageNotFound,
  QuestionDetails
};
