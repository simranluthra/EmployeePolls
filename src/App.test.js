import React from "react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { render, fireEvent } from "@testing-library/react";
import { act } from "react-dom/test-utils";

import store from "./redux";

import { _saveQuestion, _saveQuestionAnswer } from "./redux/_DATA";
import { Home, Login, NewQuestion } from "./components";

const loginProps = {
  users: {
    sarahedo: {
      id: "sarahedo",
      password: "password123",
      name: "Sarah Edo",
      avatarURL: "/avatars/sarah.jpg",
      answers: {
        "8xf0y6ziyjabvozdd253nd": "optionOne",
        "6ni6ok3ym7mf1p33lnez": "optionOne",
        am8ehyc8byjqgar0jgpub9: "optionTwo",
        loxhs1bqm25b708cmbf3g: "optionTwo",
      },
      questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
    },
  },
};

describe("1. verify _saveQuestion retuns saved ques", () => {
  it("tests all expected fields are populated", async () => {
    const mockQues = {
      optionOneText: "React",
      optionTwoText: "Node",
      author: "sarahedo",
    };
    const savedQuestion = await _saveQuestion(mockQues);
    expect(savedQuestion).toEqual(
      expect.objectContaining({
        id: expect.any(String),
        author: mockQues.author,
        optionOne: { text: mockQues.optionOneText, votes: [] },
        optionTwo: { text: mockQues.optionTwoText, votes: [] },
        timestamp: expect.any(Number),
      })
    );
  });
});

describe("2. verify _saveQuestion retuns error in case incorrect data passed", () => {
  it("retuns error in case incorrect data passed", async () => {
    const questionMissingOption1 = {
      optionTwoText: "Node",
      author: "sarah",
    };
    await expect(_saveQuestion(questionMissingOption1)).rejects.toEqual(
      "Please provide optionOneText, optionTwoText, and author"
    );
  });
});

describe("3. verify _saveQuestionAnswer returns correct ques answer", () => {
  it("returns formatted question", async () => {
    const mockAns = {
      authedUser: "sarahedo",
      qid: "8xf0y6ziyjabvozdd253nd",
      answer: "optionTwo",
    };
    const { questions, users } = await _saveQuestionAnswer(mockAns);
    expect(
      users[mockAns.authedUser].answers[mockAns.qid] === mockAns.answer
    ).toBe(true);
    expect(
      questions[mockAns.qid][mockAns.answer].votes.includes(mockAns.authedUser)
    ).toBe(true);
  });
});

describe("4. verify _saveQuestionAnswer returns error if incorrect ques answer passed", () => {
  it("returns error if incorrect ques answer passed", async () => {
    const mockIncorrectAnswer = {
      authedUser: "sarahedo",
      qid: "8xf0y6ziyjabvozdd253nd",
    };
    await expect(_saveQuestionAnswer(mockIncorrectAnswer)).rejects.toEqual(
      "Please provide authedUser, qid, and answer"
    );
  });
});

describe("5. Snapshot test", () => {
  it("tests NewQuestion page with snapshot", () => {
    const renderComponent = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewQuestion />
        </BrowserRouter>
      </Provider>
    );
    expect(renderComponent).toMatchSnapshot();
  });
});

describe("6. DOM Test", () => {
  it("tests login button text change on signing in", async () => {
    let component;
    act(() => {
      component = render(
        <React.StrictMode>
          <Provider store={store}>
            <BrowserRouter>
              <Login {...loginProps} />
            </BrowserRouter>
          </Provider>
        </React.StrictMode>
      );
    });
    const loginBtn = component.getByTestId("login-btn");
    fireEvent.click(loginBtn);
    expect(component.getByTestId("loading-text")).toHaveTextContent(
      "Signing in..."
    );
  });
});

// Remaining 4 random unit tests

describe("7. Snapshot test for home page", () => {
  const props = {
    authedUser: "sarahedo",
    questions: {
      "8xf0y6ziyjabvozdd253nd": {
        id: "8xf0y6ziyjabvozdd253nd",
        author: "sarahedo",
        timestamp: 1467166872634,
        optionOne: {
          votes: ["sarahedo"],
          text: "Build our new application with Javascript",
        },
        optionTwo: {
          votes: [],
          text: "Build our new application with Typescript",
        },
      },
      "6ni6ok3ym7mf1p33lnez": {
        id: "6ni6ok3ym7mf1p33lnez",
        author: "mtsamis",
        timestamp: 1468479767190,
        optionOne: {
          votes: [],
          text: "hire more frontend developers",
        },
        optionTwo: {
          votes: ["mtsamis", "sarahedo"],
          text: "hire more backend developers",
        },
      }
    },
    users: {
      sarahedo: {
        id: "sarahedo",
        password: "password123",
        name: "Sarah Edo",
        avatarURL: "/avatars/sarah.jpg",
        answers: {
          "8xf0y6ziyjabvozdd253nd": "optionOne",
          "6ni6ok3ym7mf1p33lnez": "optionOne",
          am8ehyc8byjqgar0jgpub9: "optionTwo",
          loxhs1bqm25b708cmbf3g: "optionTwo",
        },
        questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
      },
      mtsamis: {
        id: "mtsamis",
        password: "xyz123",
        name: "Mike Tsamis",
        avatarURL: "/avatars/mike.jpg",
        answers: {
          xj352vofupe1dqz9emx13r: "optionOne",
          vthrdm985a262al8qx3do: "optionTwo",
          "6ni6ok3ym7mf1p33lnez": "optionOne",
        },
        questions: ["6ni6ok3ym7mf1p33lnez", "xj352vofupe1dqz9emx13r"],
      }
    },
  };
  it("Home Page to match snapshot", () => {
    const renderComponent = render(
      <Provider store={store}>
        <BrowserRouter>
          <Home {...props} />
        </BrowserRouter>
      </Provider>
    );
    expect(renderComponent).toMatchSnapshot();
  });
});

describe("8. Text change test for new question page", () => {
  it("tests the notify functionality when empty form is submitted", async () => {
    let component;
    act(() => {
      component = render(
        <React.StrictMode>
          <Provider store={store}>
            <BrowserRouter>
              <NewQuestion />
            </BrowserRouter>
          </Provider>
        </React.StrictMode>
      );
    });
    const submitBtn = component.getByTestId("submit-btn");
    fireEvent.click(submitBtn);
    expect(component.getByTestId("notify-text")).toHaveTextContent(
      "Options cannot be empty."
    );
  });

  it("9. gives error when try to submit with only 1 option", async () => {
    let component;
    act(() => {
      component = render(
        <React.StrictMode>
          <Provider store={store}>
            <BrowserRouter>
              <NewQuestion />
            </BrowserRouter>
          </Provider>
        </React.StrictMode>
      );
    });
    const optionOne = component.getByTestId("option-one");
    const submitBtn = component.getByTestId("submit-btn");
    fireEvent.change(optionOne, { target: { value: "submitting with only 1 option" } });
    fireEvent.click(submitBtn);
    expect(component.getByTestId("notify-text")).toHaveTextContent(
      "Options cannot be empty."
    );
  });
});

describe("10. Login page test", () => {
  it("has fields for username, password as well as login button", async () => {
    let component;
    act(() => {
      component = render(
        <React.StrictMode>
          <Provider store={store}>
            <BrowserRouter>
              <Login {...loginProps} />
            </BrowserRouter>
          </Provider>
        </React.StrictMode>
      );
    });
    expect(component.getByTestId("input-username")).toBeTruthy()
    expect(component.getByTestId("input-password")).toBeTruthy()
    expect(component.getByTestId("login-btn")).toBeTruthy()
  });
});
