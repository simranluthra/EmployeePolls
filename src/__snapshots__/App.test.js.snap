// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`5. Snapshot test tests NewQuestion page with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mt-4 text-center"
      >
        <h1>
          Would You Rather
        </h1>
        <p>
          Create your own poll
        </p>
        <label
          class="form-label"
          for="question1"
        >
          First Option:
        </label>
        <input
          class="form-control"
          data-testid="option-one"
          placeholder="First option.."
          type="text"
          value=""
        />
        <label
          class="form-label"
          for="question2"
        >
          Second Option:
        </label>
        <input
          class="form-control"
          data-testid="option-two"
          placeholder="Second option.."
          type="text"
          value=""
        />
        <button
          class="btn btn-primary my-3"
          data-testid="submit-btn"
          type="button"
        >
          Submit
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mt-4 text-center"
    >
      <h1>
        Would You Rather
      </h1>
      <p>
        Create your own poll
      </p>
      <label
        class="form-label"
        for="question1"
      >
        First Option:
      </label>
      <input
        class="form-control"
        data-testid="option-one"
        placeholder="First option.."
        type="text"
        value=""
      />
      <label
        class="form-label"
        for="question2"
      >
        Second Option:
      </label>
      <input
        class="form-control"
        data-testid="option-two"
        placeholder="Second option.."
        type="text"
        value=""
      />
      <button
        class="btn btn-primary my-3"
        data-testid="submit-btn"
        type="button"
      >
        Submit
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`5. Snapshot tests for components: New Question Page to match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mt-4 text-center"
      >
        <h1>
          Would You Rather
        </h1>
        <p>
          Create your own poll
        </p>
        <label
          class="form-label"
          for="question1"
        >
          First Option:
        </label>
        <input
          class="form-control"
          data-testid="option-one"
          placeholder="First option.."
          type="text"
          value=""
        />
        <label
          class="form-label"
          for="question2"
        >
          Second Option:
        </label>
        <input
          class="form-control"
          data-testid="option-two"
          placeholder="Second option.."
          type="text"
          value=""
        />
        <button
          class="btn btn-primary my-3"
          data-testid="submit-btn"
          type="button"
        >
          Submit
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mt-4 text-center"
    >
      <h1>
        Would You Rather
      </h1>
      <p>
        Create your own poll
      </p>
      <label
        class="form-label"
        for="question1"
      >
        First Option:
      </label>
      <input
        class="form-control"
        data-testid="option-one"
        placeholder="First option.."
        type="text"
        value=""
      />
      <label
        class="form-label"
        for="question2"
      >
        Second Option:
      </label>
      <input
        class="form-control"
        data-testid="option-two"
        placeholder="Second option.."
        type="text"
        value=""
      />
      <button
        class="btn btn-primary my-3"
        data-testid="submit-btn"
        type="button"
      >
        Submit
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`7. Snapshot test for home page Home Page to match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="container"
      >
        <div
          class="bg-body rounded shadow-sm"
        >
          <h3
            class="text-center mt-5"
          >
            New Questions
          </h3>
          <div
            class="bg-body rounded shadow-sm"
          >
            <div
              class="card-group"
            />
          </div>
        </div>
        <div
          class="bg-body rounded shadow-sm"
        >
          <h3
            class="text-center mt-5"
          >
            Done
          </h3>
          <div
            class="bg-body rounded shadow-sm"
          >
            <div
              class="card-group"
            >
              <div
                class="mt-5 card"
              >
                <div
                  class="card-body"
                >
                  <div
                    class="card-title h5"
                  >
                    <h6
                      class="mb-0"
                    >
                      Mike Tsamis
                    </h6>
                  </div>
                  <p
                    class="card-text"
                  >
                    <small
                      class="opacity-50 text-nowrap"
                    >
                      Thu Jul 14 2016
                    </small>
                  </p>
                  <a
                    aria-current="true"
                    class="list-group-item list-group-item-action d-flex gap-3 py-3"
                    href="/questions/6ni6ok3ym7mf1p33lnez"
                  >
                    <button
                      class="btn btn-primary"
                      type="button"
                    >
                      Show
                    </button>
                  </a>
                </div>
              </div>
              <div
                class="mt-5 card"
              >
                <div
                  class="card-body"
                >
                  <div
                    class="card-title h5"
                  >
                    <h6
                      class="mb-0"
                    >
                      Sarah Edo
                    </h6>
                  </div>
                  <p
                    class="card-text"
                  >
                    <small
                      class="opacity-50 text-nowrap"
                    >
                      Wed Jun 29 2016
                    </small>
                  </p>
                  <a
                    aria-current="true"
                    class="list-group-item list-group-item-action d-flex gap-3 py-3"
                    href="/questions/8xf0y6ziyjabvozdd253nd"
                  >
                    <button
                      class="btn btn-primary"
                      type="button"
                    >
                      Show
                    </button>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="container"
    >
      <div
        class="bg-body rounded shadow-sm"
      >
        <h3
          class="text-center mt-5"
        >
          New Questions
        </h3>
        <div
          class="bg-body rounded shadow-sm"
        >
          <div
            class="card-group"
          />
        </div>
      </div>
      <div
        class="bg-body rounded shadow-sm"
      >
        <h3
          class="text-center mt-5"
        >
          Done
        </h3>
        <div
          class="bg-body rounded shadow-sm"
        >
          <div
            class="card-group"
          >
            <div
              class="mt-5 card"
            >
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  <h6
                    class="mb-0"
                  >
                    Mike Tsamis
                  </h6>
                </div>
                <p
                  class="card-text"
                >
                  <small
                    class="opacity-50 text-nowrap"
                  >
                    Thu Jul 14 2016
                  </small>
                </p>
                <a
                  aria-current="true"
                  class="list-group-item list-group-item-action d-flex gap-3 py-3"
                  href="/questions/6ni6ok3ym7mf1p33lnez"
                >
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    Show
                  </button>
                </a>
              </div>
            </div>
            <div
              class="mt-5 card"
            >
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  <h6
                    class="mb-0"
                  >
                    Sarah Edo
                  </h6>
                </div>
                <p
                  class="card-text"
                >
                  <small
                    class="opacity-50 text-nowrap"
                  >
                    Wed Jun 29 2016
                  </small>
                </p>
                <a
                  aria-current="true"
                  class="list-group-item list-group-item-action d-flex gap-3 py-3"
                  href="/questions/8xf0y6ziyjabvozdd253nd"
                >
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    Show
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`7. Snapshot tests for components: New Question Page to match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="container"
      >
        <div
          class="bg-body rounded shadow-sm"
        >
          <h3
            class="text-center mt-5"
          >
            New Questions
          </h3>
          <div
            class="bg-body rounded shadow-sm"
          >
            <div
              class="card-group"
            />
          </div>
        </div>
        <div
          class="bg-body rounded shadow-sm"
        >
          <h3
            class="text-center mt-5"
          >
            Done
          </h3>
          <div
            class="bg-body rounded shadow-sm"
          >
            <div
              class="card-group"
            >
              <div
                class="mt-5 card"
              >
                <div
                  class="card-body"
                >
                  <div
                    class="card-title h5"
                  >
                    <h6
                      class="mb-0"
                    >
                      Mike Tsamis
                    </h6>
                  </div>
                  <p
                    class="card-text"
                  >
                    <small
                      class="opacity-50 text-nowrap"
                    >
                      Thu Jul 14 2016
                    </small>
                  </p>
                  <a
                    aria-current="true"
                    class="list-group-item list-group-item-action d-flex gap-3 py-3"
                    href="/questions/6ni6ok3ym7mf1p33lnez"
                  >
                    <button
                      class="btn btn-primary"
                      type="button"
                    >
                      Show
                    </button>
                  </a>
                </div>
              </div>
              <div
                class="mt-5 card"
              >
                <div
                  class="card-body"
                >
                  <div
                    class="card-title h5"
                  >
                    <h6
                      class="mb-0"
                    >
                      Sarah Edo
                    </h6>
                  </div>
                  <p
                    class="card-text"
                  >
                    <small
                      class="opacity-50 text-nowrap"
                    >
                      Wed Jun 29 2016
                    </small>
                  </p>
                  <a
                    aria-current="true"
                    class="list-group-item list-group-item-action d-flex gap-3 py-3"
                    href="/questions/8xf0y6ziyjabvozdd253nd"
                  >
                    <button
                      class="btn btn-primary"
                      type="button"
                    >
                      Show
                    </button>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="container"
    >
      <div
        class="bg-body rounded shadow-sm"
      >
        <h3
          class="text-center mt-5"
        >
          New Questions
        </h3>
        <div
          class="bg-body rounded shadow-sm"
        >
          <div
            class="card-group"
          />
        </div>
      </div>
      <div
        class="bg-body rounded shadow-sm"
      >
        <h3
          class="text-center mt-5"
        >
          Done
        </h3>
        <div
          class="bg-body rounded shadow-sm"
        >
          <div
            class="card-group"
          >
            <div
              class="mt-5 card"
            >
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  <h6
                    class="mb-0"
                  >
                    Mike Tsamis
                  </h6>
                </div>
                <p
                  class="card-text"
                >
                  <small
                    class="opacity-50 text-nowrap"
                  >
                    Thu Jul 14 2016
                  </small>
                </p>
                <a
                  aria-current="true"
                  class="list-group-item list-group-item-action d-flex gap-3 py-3"
                  href="/questions/6ni6ok3ym7mf1p33lnez"
                >
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    Show
                  </button>
                </a>
              </div>
            </div>
            <div
              class="mt-5 card"
            >
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  <h6
                    class="mb-0"
                  >
                    Sarah Edo
                  </h6>
                </div>
                <p
                  class="card-text"
                >
                  <small
                    class="opacity-50 text-nowrap"
                  >
                    Wed Jun 29 2016
                  </small>
                </p>
                <a
                  aria-current="true"
                  class="list-group-item list-group-item-action d-flex gap-3 py-3"
                  href="/questions/8xf0y6ziyjabvozdd253nd"
                >
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    Show
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
